### PR TITLE
Call MethodHandles::unreflect in Reflection utils

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
Since unreflect calls caused GC issues recently unifying their callsites
will help us better track their usage.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/util/Reflection.java [hunk 1]
```diff
@@ -113,4 +127,22 @@ public static MethodHandle constructorMethodHandle(StandardErrorCode errorCode,
...
+
+    /**
+     * Returns a MethodHandle corresponding to the specified constructor.
+     * <p>
+     * Warning: The way Oracle JVM implements producing MethodHandle for a constructor involves
+     * creating JNI global weak references. G1 processes such references serially. As a result,
+     * calling this method in a tight loop can create significant GC pressure and significantly
+     * increase application pause time.
+     */
+    public static MethodHandle constructorMethodHandle(StandardErrorCode errorCode, Constructor constructor)
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/util/Reflection.java]: If the `unreflect` call is expensive, can we just use:
```java
    public static MethodHandle methodMethodHandle(StandardErrorCode errorCode, Method method)
    {
        try {
            return MethodHandles.lookup().findConstructor(method.getDeclaringClass(), MethodType.methodType(method.getReturnType(), method.getParameterTypes()));
        }
        catch (NoSuchMethodException | IllegalAccessException e) {
            throw new PrestoException(errorCode, e);
        }
    }

    public static MethodHandle constructorMethodHandle(StandardErrorCode errorCode, Constructor<?> constructor)
    {
        try {
            return MethodHandles.lookup().findConstructor(constructor.getDeclaringClass(), MethodType.methodType(constructor.getDeclaringClass(), constructor.getParameterTypes()));
        }
        catch (NoSuchMethodException | IllegalAccessException e) {
            throw new PrestoException(errorCode, e);
        }
    }
```

Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/util/Reflection.java]: findXXX methods have the same problem.

Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/util/Reflection.java]: Yep, confirmed that `findConstructor` also has the same problem.

Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/util/Reflection.java]: So which methods don't have this problem?

Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/util/Reflection.java]: It's hard to list all of them, but the code path that triggers this issue goes through the `MethodHandleNatives.init()` call IIRC, so you can track from there.

**Pull request merged**